src/main/java/wolox/training/config/SpringConfig.java

```diff
@@ -0,0 +1,16 @@
+package wolox.training.config;
+
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.client.RestTemplate;
+
+@Configuration
+@EnableAutoConfiguration
+public class SpringConfig {
+
+    @Bean
+    public RestTemplate restTemplate() {
```

Comment:
Why do you need this bean?

Reply:
I used this bean to, first perform calls to external API and second to be able to mock it as dependency for Unit Testing so I can emulated expected responses.

Reply:
You dont really need this bean,  rest template is already in the context, also you can mock it  in the test context, look at this [example](https://www.baeldung.com/spring-mock-rest-template)

Reply:
In the sample, checking the Github repo, they also add a Bean for the rest template. Just before oint 4 they explain: We should keep in mind that the RestTemplate in the test class should be the same instance used in the EmployeeService class. To ensure that, we have defined a RestTemplate bean in the spring config and auto-wired the instance in both test and implementation

Reply:
To do this, you can mock the service it self

Reply:
Sorry i taking in count that you are using mockito, but you are using spring test in this case you must declare the bean. You are right.

Reply:
👍